asset/asset.go

```diff
@@ -349,6 +349,22 @@ func NewSpecifierFromGroupKey(groupPubKey btcec.PublicKey) Specifier {
 	}
 }
 
+// String returns a human-readable description of the specifier.
+func (s *Specifier) String() string {
+	var assetID ID
+	s.WhenId(func(id ID) {
+		assetID = id
+	})
+
+	var groupKeyBytes []byte
+	s.WhenGroupPubKey(func(key btcec.PublicKey) {
+		groupKeyBytes = key.SerializeCompressed()
+	})
+
+	return fmt.Sprintf("AssetSpecifier(id=%s, group_pub_key=%x)",
```

Comment:
Since we now kind of have a semantic distinction between the asset being specified (`fn.Option`) or all zero (for specifying an asset is BTC), shouldn't we also make the distinction here?
Meaning that if the asset ID isn't set, we'd print `id=, group_pub_key`?